radio/src/storage/sdcard_yaml.cpp

```diff
@@ -47,23 +49,62 @@ const char * readYamlFile(const char* fullpath, const YamlParserCalls* calls, vo
     if (result != FR_OK) {
         return SDCARD_ERROR(result);
     }
-      
+
     YamlParser yp; //TODO: move to re-usable buffer
     yp.init(calls, parser_ctx);
 
+    uint16_t calculated_checksum = 0xFFFF;
+    uint16_t file_checksum = 0;
+
     char buffer[32];
     while (f_read(&file, buffer, sizeof(buffer), &bytes_read) == FR_OK) {
 
       // reached EOF?
       if (bytes_read == 0)
         break;
-      
+
+      // Skip the 'checkum' value in the start of the file if present
+      uint16_t skip = 0;
+      if(strncmp(buffer, "checksum: ", 10) == 0) {
+        skip = 10;
+        char* startPos = buffer + 10; // length of "checksum: "
+        char* endPos = startPos;
+        while(*endPos != '\r') {
+          if (endPos > buffer + bytes_read) {
+             return SDCARD_ERROR(	FR_INT_ERR);
+          }
+          endPos++;
+        }
+        *endPos = 0;
+
+        // Skip LF of line-ending CR-LF pair
+        endPos++;
+        if(*endPos == '\n') {
+          endPos++;
+        }
+        file_checksum = atoi(startPos);
+        skip = endPos - buffer;
+      }
+
```

Comment:
This should be done only once and not on every read. When longer texts are allowed combined with some bad luck, "checksum: " can be part of another config and by chance be at the beginning of a buffer.
I would prefer a read before you enter the while loop or at least a bool to only check for the checksum at the first run.

Reply:
Yeah, absolutely right - I will update the PR

Reply:
I added a flag to have the check only run once. I preferred that over having two read blocks of the code - which seemed to fragment the read - and I think it also produces a larger binary - however I am willing to be proven wrong :-)